leapp/reporting/__init__.py

```diff
@@ -1,106 +1,260 @@
+import enum
 import json
 
 from leapp.models import fields, Model
 from leapp.topics import ReportTopic
 from leapp.libraries.stdlib.api import produce
 
 
-class Renderers(Model):
+class Report(Model):
     """
-    Model for describing renderers to be used when presenting reports.
+    Framework model used for reporting
     """
 
+    __non_inheritable__ = True
+
     topic = ReportTopic
 
-    html = fields.String()
-    """
-    Render report message as html
-    """
-    plaintext = fields.String()
-    """
-    Render report message as plaintext
-    """
+    report = fields.JSON()
 
 
-class Report(Model):
+class PrimitiveType(enum.Enum):
     """
-    Framework model used for reporting and presentation (see "renderers" field) purposes. The report can also carry
-    a special meaning using "flags" field.
+    Boolean to set whether the primitive leaf is supposed to be list or dict
     """
+    scalar = 0
+    vector = 1
 
-    __non_inheritable__ = True
 
-    topic = ReportTopic
+def _add(data, path, value, leaf_list=False):
```

Comment:
I really miss some basic unit tests for that. At least to make out on the spot what arrives as path.

Reply:
tests added